### PR TITLE
feat: Support Event Source Mapping `metrics_config`, `provisioned_poller_config`, and Lambda Recursion Loop

### PR DESCRIPTION
**Pull request by magreenbaum**
## Description
Supports arguments `aws_lambda_event_source_mapping.metrics_config`, `aws_lambda_event_source_mapping.provisioned_poller_config` and resource for `aws_lambda_function_recursion_config`. 

## Motivation and Context
- https://github.com/hashicorp/terraform-provider-aws/pull/40322 
- https://github.com/hashicorp/terraform-provider-aws/issues/40303
- Closes: https://github.com/terraform-aws-modules/terraform-aws-lambda/issues/643

## Breaking Changes
No.

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -432,6 +448,17 @@ resource "aws_lambda_function_url" "this" {
   }
 }
 
+resource "aws_lambda_function_recursion_config" "this" {
+  count = local.create && var.recursive_loop == "Allow" ? 1 : 0
+
+  function_name  = var.function_name
```
Comment (antonbabenko) [inline main.tf:454]:
```suggestion
  function_name  = aws_lambda_function.this[0].function_name
```

Then we don't need `depends_on`.

File: main.tf [hunk 2]
```diff
@@ -432,6 +448,17 @@ resource "aws_lambda_function_url" "this" {
   }
 }
 
+resource "aws_lambda_function_recursion_config" "this" {
+  count = local.create && var.recursive_loop == "Allow" ? 1 : 0
```
Comment (antonbabenko) [inline main.tf:452]:
```suggestion
  count = local.create && var.create_function && !var.create_layer && var.recursive_loop == "Allow" ? 1 : 0
```

**Review by antonbabenko — CHANGES_REQUESTED** [review]

**Review by antonbabenko — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 7.17.0 :tada:

File: examples/alias/versions.tf [hunk 3]
```diff
@@ -4,7 +4,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 5.70"
+      version = ">= 5.79"
```
Comment (ikuwow) [inline examples/alias/versions.tf:7]: This is a breaking change.
This changes should be noted in CHANGELOG.md.
https://github.com/terraform-aws-modules/terraform-aws-lambda/blob/master/CHANGELOG.md#7170-2024-12-08

File: versions.tf [hunk 4]
```diff
@@ -4,7 +4,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 5.70"
+      version = ">= 5.79"
```
Comment (ikuwow) [inline versions.tf:7]: This is a breaking change.
This changes should be noted in CHANGELOG.md.
https://github.com/terraform-aws-modules/terraform-aws-lambda/blob/master/CHANGELOG.md#7170-2024-12-08

**Pull request merged**